docs/apis/batch/libs/ml/cross_validation.md

```diff
@@ -0,0 +1,175 @@
+
+## Cross Validation Strategies
+
+There are several strategies for holding out data. FlinkML has convenience methods for
+- Train-Test Splits
+- Train-Test-Holdout Splits
+- K-Fold Splits
+- Multi-Random Splits
+
+### Train-Test Splits
+
+The simplest method of splitting is the `trainTestSplit`. This split takes a DataSet and a parameter *fraction*.  The *fraction* indicates the portion of the DataSet that should be allocated to the training set. This split also takes two additional optional parameters, *precise* and *seed*.  
+
+By default, the Split is done by randomly deciding weather or not an observation is assigned to the training DataSet with probability = *fraction*.  When *precise* is `true` however, additional steps are taken to ensure the training set is as close as possible to the length of the DataSet  $\cdot$ *fraction*.
+
+The method returns a new `TrainTestDataSet` object which has a `.training` attribute containing the training DataSet and a `.testing` attribute containing the testing DataSet.
+
+
+### Train-Test-Holdout Splits
+
+In some cases, algorithms have been known to 'learn' the testing set.  To combat this issue, a train-test-hold out strategy introduces a secondary holdout set, aptly called the *holdout* set.
+
+Traditionally, training and testing would be done to train an algorithms as normal and then a final test of the algorithm on the holdout set would be done.  Ideally, prediction errors/model scores in the holdout set would not be significantly different than those observed in the testing set.
+
+In a train-test-holdout strategy we sacrifice the sample size of the initial fitting algorithm for increased confidence that our model is not over-fit.
+
+When using `trainTestHoldout` splitter, the *fraction* `Double` is replaced by a *fraction* array of length three. The first element coresponds to the portion to be used for training, second for testing, and third for holdout.  The weights of this array are *relative*, e.g. an array `Array(3.0, 2.0, 1.0)` would results in approximately 50% of the observations being in the training set, 33% of the observations in the testing set, and 17% of the observations in holdout set.
+
+### K-Fold Splits
+
+In a *k-fold* strategy, the DataSet is split into *k* equal subsets. Then for each of the *k* subsets, a `TrainTestDataSet` is created where the subset is the `.training` DataSet, and the remaining subsets are the `.testing` set.
+
+For each training set, an algorithm is trained and then is evaluated based on the predictions based on the assosciated testing set. When an algorithm that has consistent grades (e.g. prediction errors) across held out datasets we can have some confidence that our approach (e.g. choice of algorithm / algorithm parameters / number of iterations) is robust against overfitting.
+
+<a href="https://en.wikipedia.org/wiki/Cross-validation_(statistics)#k-fold_cross-validation">K-Fold Cross Validatation</a>
```

Comment:
It's been a while since I wrote this, but I vaguely remember having some sort of issue either with the build (or just my mark down interpretter) specifically on this link, I think it had to do with the parentheses in the link? That's why I fell back to html.  Also, I never can get markdown links to come out correctly in tables.